lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -11,671 +12,443 @@
+    "auto" = "SERVICE_AUTO_START"
+    "delayed" = "SERVICE_DELAYED_AUTO_START"
+    "manual" = "SERVICE_DEMAND_START"
+    "disabled" = "SERVICE_DISABLED"
 }
 
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","started","stopped","restarted" -resultobj $result
+$display_name = Get-AnsibleParam -obj $params -name 'display_name' -type 'str'
+$description = Get-AnsibleParam -obj $params -name 'description' -type 'str'
 
-$application = Get-AnsibleParam -obj $params -name "application" -type "str"
+$application = Get-AnsibleParam -obj $params -name "application" -type "path"
+$appDirectory  = Get-AnsibleParam -obj $params -name "working_directory" -aliases "app_directory","chdir" -type "path"
 $appParameters = Get-AnsibleParam -obj $params -name "app_parameters"
-$appParametersFree  = Get-AnsibleParam -obj $params -name "app_parameters_free_form" -type "str"
-$startMode = Get-AnsibleParam -obj $params -name "start_mode" -type "str" -default "auto" -validateset "auto","delayed","manual","disabled" -resultobj $result
+$appArguments = Get-AnsibleParam -obj $params -name "arguments" -aliases "app_parameters_free_form"
+$startMode = Get-AnsibleParam -obj $params -name "start_mode" -type "str" -default "auto" -validateset $start_modes_map.Keys -resultobj $result
 
-$stdoutFile = Get-AnsibleParam -obj $params -name "stdout_file" -type "str"
-$stderrFile = Get-AnsibleParam -obj $params -name "stderr_file" -type "str"
+$stdoutFile = Get-AnsibleParam -obj $params -name "stdout_file" -type "path"
+$stderrFile = Get-AnsibleParam -obj $params -name "stderr_file" -type "path"
 $dependencies = Get-AnsibleParam -obj $params -name "dependencies" -type "list"
 
 $user = Get-AnsibleParam -obj $params -name "user" -type "str"
 $password = Get-AnsibleParam -obj $params -name "password" -type "str"
 
-if (($appParameters -ne $null) -and ($appParametersFree -ne $null))
-{
-    Fail-Json $result "Use either app_parameters or app_parameteres_free_form, but not both"
-}
-if (($appParameters -ne $null) -and ($appParameters -isnot [string])) {
-    Fail-Json -obj $result -message "The app_parameters parameter must be a string representing a dictionary."
+$result = @{
+    changed = $false
 }
 
-Function Nssm-Invoke
-{
+function Invoke-NssmCommand {
     [CmdletBinding()]
     param(
-        [Parameter(Mandatory=$true)]
-        [string]$cmd
+        [Parameter(Mandatory=$true,ValueFromRemainingArguments=$true)]
+        [string[]]$arguments
     )
 
-    $nssm_result = Run-Command -command "nssm $cmd"
+    $executable = "nssm"
+    $argument_string = Argv-ToString -arguments $arguments
 
-    return $nssm_result
-}
+    $command = "$executable $argument_string"
```

Comment:
It would be better actually, but the command line without the path of the executable (`$argument_string`) is also returned to the caller and added to the module result object (`result.nssm_error_cmd`).
That said, I'm not sure that this return value is really useful for the user, maybe it could be removed. This is one of the points of the rework to do on the values returned by the module.